### PR TITLE
Add community's tag naming conventions to javadoc

### PR DESCRIPTION
**Pull request by Virtuoel**
Effectively a placeholder that can be replaced by a decided convention in the future.

### CODE REVIEW

File: fabric-tag-extensions-v0/src/main/java/net/fabricmc/fabric/api/tag/TagRegistry.java [hunk 1]
```diff
@@ -34,6 +34,10 @@
 
 /**
  * Helper methods for registering Tags.
+ *
+ * <p>The convention for namespaces of common tags (e.g. ingots) is {@code c}.
+ *
+ * <p>The convention for paths is plural terms, e.g. {@code c:iron_ingots}, following the majority of vanilla's tags, e.g. {@code minecraft:slabs}, {@code minecraft:gold_ores}
```
Comment (haykam821) [inline fabric-tag-extensions-v0/src/main/java/net/fabricmc/fabric/api/tag/TagRegistry.java]:
```suggestion
 * <p>The convention for paths is to use plural terms, such as {@code c:iron_ingots}, keeping in line with the majority of vanilla tags such as {@code minecraft:slabs} and {@code minecraft:gold_ores}.
```

**Review by haykam821 — CHANGES_REQUESTED** [review]

**Comment by TheBrokenRail** [comment]
Maybe ``c:ingots/iron`` instead of ``c:iron_ingots`` so you aren't stuck with a giant tags folder.

**Comment by haykam821** [comment]
That isn't convention..

**Comment by TheBrokenRail** [comment]
Isn't this PR literally defining convention, rather than just picking what other mods use?

**Comment by haykam821** [comment]
It wouldn't be convention then.

**Comment by TheBrokenRail** [comment]
Convention can change, especially if there is a single standard.

**Comment by haykam821** [comment]
Usually convention changes in response to a new implementation, rather than documentation.

**Comment by TheBrokenRail** [comment]
Yes, but most implementations will change to what FAPI says

**Comment by haykam821** [comment]
Most changes to Fabric API are agreed upon, and `c:ingots/iron` is not.

**Comment by TheBrokenRail** [comment]
That is why this is a suggestion sin a PR

**Comment by i509VCB** [comment]
Convention changes would usually go through a RFC. Clearly we have gone for a PR the standard in and hope no one notices

**Comment by liach** [comment]
I wonder if player and modmuss want to adopt this. If we add this to fabric api it means fabricmc organization officially back this convention.

**Comment by Virtuoel** [comment]
Modified to be a placeholder that can be updated if/when conventions are decided upon.

**Comment by shartte** [comment]
Big 👎 from me. Convention namespace is clearly stated on the _official_ Wiki, and it is "c:".

**Comment by natanfudge** [comment]
> Modified to be a placeholder that can be updated if/when conventions are decided upon.

Well in that case the PR is pointless. What's the problem with saying the convention is `c`? it definitely won't be anything else, since many mods already use `c`.

**Comment by Virtuoel** [comment]
I'm personally fine with `c:`, as are many mods as can be seen from tag usage among mods, but as liach mentioned, are modmuss, player, i5, other triage members, etc. fine with adopting the community's current tag namespace and path naming conventions (`c:` and plurals) as *the* conventions endorsed by official javadocs?
If so, great! Just tell me and I'll gladly revert the placeholders so we can finally get a convention that's not just on the wiki.

**Comment by Earthcomputer** [comment]
If you are a modder learning Fabric, what you want is your mod to *work* with other mods, and there is 1 convention that is vastly more common than any other. Would you really give a fuck about what Fabric officially endorses? Javadocs should be *useful*, these Javadocs are not useful at all, they only mislead the reader away from the frankly more important knowledge that `c:` is the most common convention. If you are going to put Javadocs here at all, best say something like:
```
Although not officially endorsed by Fabric, most mods use {@code c:} as the namespace for common things.
```

**Comment by TheBrokenRail** [comment]
IMO if an standard is endorsed by Fabric, most existing big resource mods (cotton resources, etc) would probably switch to it quickly, considering its just a bunch of renaming. Thus making it the standard new modders would want to learn.

**Comment by Virtuoel** [comment]
Updated to state the conventions again, now clarifying that the stated conventions are not official.

**Review by Earthcomputer — APPROVED** [review]

**Comment by Earthcomputer** [comment]
This looks much better. No javadoc was better than the old javadoc.

**Review by liach — CHANGES_REQUESTED** [review]
Imo we should just provide a link to the wiki's tag guide. That guide can update while the api is known to be really sluggish at updating.

**Comment by ghost** [comment]
Because this PR kind of touches on a greater discourse in the fabric community I think it would be a really good idea to start an RFC about tag naming conventions (as @i509VCB mentioned) and put this PR on hold given there's been a lot of debate surrounding the topic. It seems like everyone has their own opinion on how we should proceed in the future, and so actually giving the community a chance to have an open dialog about the issue might be beneficial. My personal opinion is that the current "c" namespace is kind of outdated, it grew out of necessity and not actual standard, and there's not a good consensus on how things ought to be named within that namespace (going back to the whole discourse on whether it should be c:ingots/iron or c:iron_ingots or c:iron_ingot), so making a PR for fabric to "officially" endorse a system that is ambiguous seems reckless. I feel like what we need is more of a "de jure" standard for how a common namespace ought to work not a "de facto" norm, and the best way to do that would be through a formal RFC.

Personally I think using something like "common" for the common namespace is a lot more intuitive as to what the namespace is actually for than the "c". Using just one letter isn't helpful to new people trying to mod, or people from outside the community trying to make datapacks that work with fabric. I think someone mentioned that outside of fabric "g" is the standard and I literally had to think for a minute about what "g" could stand for. Having a common namespace as just one letter is not intuitive at all. However, that doesn't really matter right now, the point is that this kind of discussion should be happening in a more public and official setting, not a PR that puts the cart before the horse.

**Comment by liach** [comment]
I've seen a data pack community declaring `global` as the common namespace. Imo whatever it is we should put it on wiki.

**Comment by franiscoder** [comment]
I don't like "c" is the common namespace but I also think that fabric can and should take this as a way to show fabric mod compatibility isn't that bad. 
This PR is just a javadoc, it's already not officially supporting it but just a blob of important information.

**Review by frqnny — COMMENTED** [review]
I don't like "c" is the common namespace but I also think that fabric can and should take this as a way to show fabric mod compatibility isn't that bad. 
This PR is just a javadoc, it's already not officially supporting it but just a blob of important information.

**Comment by Technici4n** [comment]
I think having it on the official wiki is good enough.
[The wiki page](https://fabricmc.net/wiki/tutorial:tags) contains much more information than we could put in javadoc anyway, especially given that the wiki lists tags already used by other mods.
For example, if you are hesitating between `aluminum` and `aluminium`, you should use `c:aluminum_*` to ensure Tech Reborn compatibility even if you think it should be called `c:aluminium_*`.

As such, I am in favor of closing this PR.

**Comment by liach** [comment]
This pr should really be edited to add something like

```java
/**
 * @see https://fabricmc.net/wiki/tutorial:tags Tags tutorial on the Fabric Wiki
 */
```

**Comment by Technici4n** [comment]
I think the new convention tags module properly defines the standard now.

**Pull request closed without merging**